provisioner/terraform/resources.go

```diff
@@ -310,8 +310,8 @@ func applyAutomaticInstanceID(resource *tfjson.StateResource, agents []*proto.Ag
 	key, isValid := map[string]string{
 		"google_compute_instance":         "instance_id",
 		"aws_instance":                    "id",
-		"azurerm_linux_virtual_machine":   "id",
-		"azurerm_windows_virtual_machine": "id",
+		"azurerm_linux_virtual_machine":   "virtual_machine_id",
+		"azurerm_windows_virtual_machine": "virtual_machine_id",
```

Comment:
I'm actually using `azurerm_virtual_machine` in the example template because `azurerm_linux_virtual_machine` doesn't support attaching the data disk before booting, meaning I can't format it in cloud-init

Reply:
Ahh, I'll add that as a resource as well.

Reply:
Spike and I discussed, and it doesn't seem that the deprecated `azurerm_virtual_machine` resource exposes the machine ID.